### PR TITLE
Apigateway stage method settings for CloudWatch.

### PR DESCRIPTION
**Pull request by jtsaito**
This implementation adds a `method_settings` attribute set to the AWS API Gateway stage resource.

**Example**
```
resource "aws_api_gateway_stage" "api-gateway" {
  deployment_id = "ejo42b"
  rest_api_id   = "1czld7sqwj"
  stage_name    = "jtstest3"

  method_settings {
    path               = "/*/*"
    metrics_enabled    = true
    logging_level      = "ERROR"
    data_trace_enabled = true
  }
}
```

*Scope*
This PR is restricted in the following ways.

1. The attributes are ignored in the create operation (thus currently applying twice when created with the the api gateway
2. Only support method settings related to CloudWatch
3. No tests
4. No validations for the method setting values

The current feature set is a minimal implementation for adapting Babbel's existing API Gateways. Please refer to the Trello card for discussing the scope of the story (as opposed to the scope of this PR).

[Trello](https://trello.com/c/jdHh2WRX/486-3-add-missing-attributes-for-aws-api-gateway-stages)

### CODE REVIEW

File: builtin/providers/aws/resource_aws_api_gateway_stage.go [hunk 1]
```diff
@@ -44,6 +44,30 @@ func resourceAwsApiGatewayStage() *schema.Resource {
...
+						},
+						"logging_level": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"metrics_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"path": {
```
Comment (jansiwy) [inline builtin/providers/aws/resource_aws_api_gateway_stage.go]: as discussed: `method_setting_key` as called in http://docs.aws.amazon.com/apigateway/api-reference/resource/stage/

File: builtin/providers/aws/resource_aws_api_gateway_stage.go [hunk 2]
```diff
@@ -170,12 +194,35 @@ func resourceAwsApiGatewayStageRead(d *schema.ResourceData, meta interface{}) er
 
 	d.Set("deployment_id", stage.DeploymentId)
 	d.Set("description", stage.Description)
+	d.Set("method_settings", readMethodSettingsMap(d, stage))
```
Comment (jansiwy) [inline builtin/providers/aws/resource_aws_api_gateway_stage.go]: `d` not used

File: builtin/providers/aws/resource_aws_api_gateway_stage.go [hunk 3]
```diff
@@ -340,3 +393,67 @@ func resourceAwsApiGatewayStageDelete(d *schema.ResourceData, meta interface{})
...
+			})
+		}
+	}
+
+	// method settings to be updated
+	for _, newVariables := range newVars {
+		newMethodSettings := newVariables.(map[string]interface{})
+
+		if newMethodSettings["data_trace_enabled"] != nil {
+			path := newMethodSettings["path"]
```
Comment (jansiwy) [inline builtin/providers/aws/resource_aws_api_gateway_stage.go]: extract path, a.k.a. method_setting_key

**Comment by jtsaito** [comment]
Turns out the method settings already exist :( https://www.terraform.io/docs/providers/aws/r/api_gateway_method_settings.html

**Pull request closed without merging**